### ISSUE TITLE
ssh: could not start shell error when trying to connect to node

### ISSUE THREAD

**Issue by Videothek** [gravitational/teleport#11161]
### Description

**What happened**: When i try to connect to my node it says `ssh: could not start shell`

**What you expected to happen**: It opens the ssh connection to the node

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Open Teleport
2. Try to connect via Web Console

### Server Details
- Teleport version (run `teleport version`): Teleport v9.0.0 git:v9.0.0-0-g1fa8857aa go1.17.7
- Server OS (e.g. from `/etc/os-release`): Ubuntu 20.04
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): VM
- Additional details: Using teleport internaly 

### Client Details
- Tsh version (`tsh version`): Teleport v9.0.0 git:v9.0.0-0-g1fa8857aa go1.17.7
- Computer OS (e.g. Linux, macOS, Windows): Ubuntu Server 20.04
- Browser version (for UI-related issues): Firefox
- Installed via (e.g. apt, yum, brew, website download): apt
- Additional details: None

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)

**Comment by russjones** [comment]
@Videothek We'll need some more information. Can you provide some debug logs from the Teleport node or proxy to show us what is happening when you try and connect?

**Comment by Videothek** [comment]
@russjones sure, wich logs do you need?
Maybe you could tell me where they are that i can provide them here.

Thank you.

**Comment by russjones** [comment]
@Videothek Location of the logs depends on how you are running Teleport.

If you are using our systemd unit file, you can obtain them by running `journalctl -u teleport -f`.

**Comment by Videothek** [comment]
@russjones thank you for providing the necessary command.

This is what the teleport spits out:

```
Mar 19 17:57:29 teleport teleport[1198]: unknown method CreateSessionTracker for service proto.AuthService
Mar 19 17:57:29 teleport teleport[1198]: 2022-03-19T17:57:29+01:00 WARN [WEBSOCKET] Unable to stream terminal: ssh: could not start shell. web/terminal.go:496
Mar 19 17:57:29 teleport teleport[1198]: 2022-03-19T17:57:29+01:00 INFO [AUDIT]     session.data addr.remote:192.168.1.135:50989 cluster_name:teleport code:T2006I ei:2.147483646e+09 event:session.data login:usernamespace:default rx:4086 server_id:5edec3cf-86b4-4506-ae95-7b7d50463f36 sid: time:2022-03-19T16:57:29.747Z tx:4779 uid:ed98ad49-94c2-4696-8f62-6ee73ce3ac22 user:User events/emitter.go:325
```

**Comment by russjones** [comment]
@Videothek It looks like you upgraded your nodes to Teleport 9.0 without upgrading Teleport Auth Server. Can you upgrade Auth Server and retry?

@xacrimon @zmb3  I think we should add a graceful fallback to Teleport 9.0 nodes. If you can't find `CreateSessionTracker` fallback to the old logic.

**Comment by Videothek** [comment]
@russjones ok interesting, how can i upgrade the auth server?

(I installed teleport via apt)

**Comment by russjones** [comment]
@Videothek Download and install the Teleport 9.0.0 deb package. You can check the binary has been updated by running `teleport version`.

Once the latest binary is on disk, run `systemctl restart teleport`.

**Comment by Videothek** [comment]
@russjones that has fixed it, thank you very much!

**Comment by russjones** [comment]
Going to reopen this issue so we can track fixing this issue allowing 9.0 nodes to join a 8.x cluster.

**Comment by bukasbirthdayko** [comment]
Hi all, I've installed the teleport cluster on kubernetes - how do I update the auth server? is there a latest helm package available that comes with the updated auth server?

**Comment by zmb3** [comment]
> @xacrimon @zmb3 This has happened with a few users now, I think we should add a graceful fallback to Teleport 9.0 nodes. If a node can't find the `CreateSessionTracker` RPC, fallback to the old logic assuming that the node connected to an old Auth Server.

Not sure I agree. Supporting nodes newer than proxy/auth is not something Teleport has ever done. If we support it in some use cases but not others, we send a confusing message to users. If we want to support this use case, then we should commit to it, document it, add it to the test plan, etc.

Of all the people that I've seen run into this issue, none of them have deliberately wanted to run 9.0 nodes on 8.x auth - in all cases they _thought_ they upgraded auth and didn't realize they need to restart the service to run the new binary.

Adding a fallback feels like treating the symptom without considering the root of the issue - a poor upgrade process. Further, this would just hide the error and allow the user to run in an unsupported configuration without knowing anything is wrong. Lastly, there's a potential security issue here where users think they've upgraded and set things up to require moderation, but since we hide the error the session proceeds without requiring a moderator, bypassing the security checks.

tl;dr - I agree that people are getting confused and that we should do something about it. I just think we should address the source of the confusion and not swallow the error.

**Comment by bkcsfi** [comment]
Is there any documentation that describes the version requirements for the auth server vs nodes?

that is, what combinations of auth server version vs node version?

I'm trying to understand how to maintain a cluster w/o getting locked out of remote nodes because they won't connect to a newer  auth server, or an auth server won't connected to a 'newer' node.

e.g. I have a cluster now running 8.2.0 auth server. Some nodes in the cluster aren't officially supported so I had to manually compile go, and then compile teleport 8.0.0 on those systems.  I then ran into a case where upgrading the auth server to version 9 made the version 8 nodes unreachable. To avoid that happening again  I pinned dpkg to  version 8.2.0 on the auth server 

Today I updated a fully supported Ubuntu 20 node to teleport 9.0.4, but  that node doesn't work with the 8.2.0 auth server.

Fortunately I am able to reach that node through other means, but I can't downgrade it to 8.2.0 because that package is not available on teleport server anymore. If I upgrade the auth server to 9.0.4, I think my hand-built 8.0.0 nodes won't work anymore.

I understand the code needs to evolve over time, and node/auth servers must be "kind of close" version-wise, but how close?  

In what order should components be updated?  e.g. is a version 8 auth server supposed to be able to work with a version 7 node?   Are nodes never allowed to have a later version than the auth server?

w/o this guidance its difficult to be see how a person could reliably maintain a teleport cluster.  

thanks

**Comment by xacrimon** [comment]
@bkcsfi I lack a good documentation link now but the gist of it is that the upgrade pattern is auth -> proxy -> nodes in that order. Proxy and nodes should always support one major version higher of the auth server but not the other way around. Here we've decided to simply improve the error message and close the issue as is when that PR merges. So v8 nodes and proxy should support a v9 auth, if you're having issues with this, please open an issue or check with us on the community Slack.

**Comment by zmb3** [comment]
This is mostly covered here: https://goteleport.com/docs/setup/operations/upgrading/#component-compatibility

tl;dr: for node on major version N, auth/proxy must be at major version N or N+1.
